autoendpoint/src/server.rs

```diff
@@ -1,38 +1,38 @@
 //! Main application server
+#![forbid(unsafe_code)]
+use std::sync::Arc;
+use std::time::Duration;
 
-use crate::db::client::{DbClient, DbClientImpl};
-use crate::error::{ApiError, ApiResult};
-use crate::metrics;
-use crate::middleware::sentry::sentry_middleware;
-use crate::routers::adm::router::AdmRouter;
-use crate::routers::apns::router::ApnsRouter;
-use crate::routers::fcm::router::FcmRouter;
-use crate::routes::health::{
-    health_route, lb_heartbeat_route, log_check, status_route, version_route,
-};
-use crate::routes::registration::{
-    get_channels_route, new_channel_route, register_uaid_route, unregister_channel_route,
-    unregister_user_route, update_token_route,
-};
-use crate::routes::webpush::{delete_notification_route, webpush_route};
-use crate::settings::Settings;
 use actix_cors::Cors;
 use actix_web::{
-    dev, http::StatusCode, middleware::errhandlers::ErrorHandlers, web, App, FromRequest,
-    HttpServer,
+    dev, http::StatusCode, middleware::ErrorHandlers, web, web::Data, App, HttpServer,
 };
 use cadence::StatsdClient;
 use fernet::MultiFernet;
-use std::sync::Arc;
-use std::time::Duration;
+use serde_json::json;
+
+use autopush_common::db::{client::DbClient, dynamodb::DdbClientImpl, DbSettings, StorageType};
+
+use crate::error::{ApiError, ApiErrorKind, ApiResult};
+use crate::metrics;
+use crate::routers::{adm::router::AdmRouter, apns::router::ApnsRouter, fcm::router::FcmRouter};
+use crate::routes::{
+    health::{health_route, lb_heartbeat_route, log_check, status_route, version_route},
+    registration::{
+        get_channels_route, new_channel_route, register_uaid_route, unregister_channel_route,
+        unregister_user_route, update_token_route,
+    },
+    webpush::{delete_notification_route, webpush_route},
+};
+use crate::settings::Settings;
 
 #[derive(Clone)]
-pub struct ServerState {
+pub struct ServerOptions {
```

Comment:
nit: I know the goal of this rename is to closer match autoconnect but IMHO I don't think `ServerOptions` is the right name for this (I think we originally got `ServerState` from actix-web docs, looks like they now commonly call this kind of thing `AppState` as there's one per every `App` instance created). E.g. a `reqwest` client isn't really an "option".

Also looking at autconnect-settings, I think ideally we would have only a `Settings` type without the second `ServerOptions` for deserializing things from `Settings` -- remember merino accomplished this with that fancy serde library to deserialize on the fly directly into its Settings objects: https://github.com/mozilla-services/merino/blob/main/merino-settings/src/lib.rs#L200